source/beamline/BioXAS/BioXASBeamline.cpp

```diff
@@ -88,6 +117,45 @@ AMAction3* BioXASBeamline::createScanInitializationAction(AMGenericStepScanConfi
 		if (BioXASBeamlineSupport::usingScaler(configuration)) {
 			scalerInitialization = new AMListAction3(new AMListActionInfo3("BioXAS scaler initialization", "BioXAS scaler initialization"));
 			scalerInitialization->addSubAction(scaler->createContinuousEnableAction3(false)); // Check that the scaler is in single shot mode and is not acquiring.
+
+			// Determine the longest region time for the scan.
+
+			double maxRegionTime = configuration->scanAxisAt(0)->regionAt(0)->regionTime();
+
+			for (int axisIndex = 0, axisCount = configuration->scanAxes().count(); axisIndex < axisCount; axisIndex++) {
```

Comment:
Are you thinking that the max time should be a cached value, or would the loop just be shuffled over to the AMScanAxis class?